shared/wallets/wallet/settings/popups/really-remove-account/container.js

```diff
@@ -28,12 +29,16 @@ const mapDispatchToProps = (dispatch, {navigateUp}) => ({
     dispatch(navigateUp())
   },
   _onCopyKey: (secretKey: string) => dispatch(ConfigGen.createCopyToClipboard({text: secretKey})),
-  _onFinish: (accountID: Types.AccountID) =>
+  _onFinish: (accountID: Types.AccountID) => {
     dispatch(
       WalletsGen.createDeleteAccount({
         accountID,
       })
-    ),
+    )
+    if (flags.useNewRouter) {
+      dispatch(navigateUp())
```

Comment:
The problem here is that this dialog is on top of the remove-account screen, so navigating up brings us back to the remove-account screen instead of the wallet list.

What should I do here to dismiss all dialogs?

Reply:
there is an action to clear all modals

Reply:
routetreegen
```
    "clearModals": {
      "_description": "ONLY used by the new nav. use this to clear any modal routes"
    }

```

Reply:
Hmm...this doesn't seem to work. The visible path used by clearModals seems to only see the topmost modal if there are two.

Did two navigateUps instead, which seems to work.

Reply:
Hmm, might be better to fix clearModals like in https://github.com/keybase/client/pull/16506/files first

Reply:
yeah, you can wait or just copy that over

Reply:
Danny copied it over, so I changed this to use it.